pkgs/development/compilers/opensycl/default.nix

```diff
@@ -43,11 +50,17 @@ stdenv.mkDerivation rec {
   ] ++ lib.optionals rocmSupport [
     rocmPackages.clr
     rocmPackages.rocm-runtime
+  ] ++ lib.optionals cudaSupport [
+    cudaPackages.cuda_cudart
+    (lib.getOutput "stubs" cudaPackages.cuda_cudart)
   ];
 
   # opensycl makes use of clangs internal headers. Its cmake does not successfully discover them automatically on nixos, so we supply the path manually
   cmakeFlags = [
     "-DCLANG_INCLUDE_PATH=${llvmPackages_15.libclang.dev}/include"
+    (lib.cmakeBool "WITH_CUDA_BACKEND" cudaSupport)
+    (lib.cmakeBool "WITH_ROCM_BACKEND" rocmSupport)
+    (lib.cmakeBool "WITH_OPENCL_BACKEND" openclSupport)
```

Comment:
Ah, just realized this doesn't exist before https://github.com/NixOS/nixpkgs/pull/295845